lambdas/generate-gro-xml/src/test/java/GenerateGroXmlTest.java

```diff
@@ -0,0 +1,112 @@
+    private static final Context context = mock(Context.class);
+    private static final GenerateGroXml underTest = new GenerateGroXml(
+        awsService,
+        config,
+        objectMapper,
+        xmlMapper
+    );
+    private static final String bucketName = "groIngestionBucketName";
+
+    @BeforeAll
+    static void setup() {
+        doNothing().when(awsService).putInBucket(any(), any(), any());
+        when(config.getGroIngestionBucketName()).thenReturn(bucketName);
+    }
+
+    @BeforeEach
+    void refreshSetup() {
+        clearInvocations(awsService);
+        clearInvocations(config);
+        reset(xmlMapper);
+    }
+
+    @Test
+    void constructionCallsCorrectInstantiation() {
+        try (var awsService = mockConstruction(AwsService.class);
+             var config = mockConstruction(Config.class)) {
+            var mapper = mockStatic(Mapper.class);
+            new GenerateGroXml();
+            assertEquals(1, awsService.constructed().size());
+            assertEquals(1, config.constructed().size());
+            mapper.verify(Mapper::objectMapper, times(1));
+            mapper.verify(Mapper::xmlMapper, times(1));
+            mapper.close();
+        }
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = {1, 5, 10, 80})
+    void generateDeathRegistrationGroupXmlGeneratesAGivenNumberOfRecords(int numberOfRecords) throws JsonProcessingException {
+        var event = new InsertDeathXmlRequest("Test", numberOfRecords);
+        when(xmlMapper.writeValueAsString(any())).thenReturn("XmlFile");
+
+        underTest.handleRequest(event, context);
+
+        verify(xmlMapper).writeValueAsString(argThat(a -> {
+            var group = (DeathRegistrationGroup) a;
+            return group.deathRegistrations().size() == numberOfRecords;
+        }));
+        verify(awsService).putInBucket(eq(bucketName), any(), eq("XmlFile"));
+    }
+
+    @Test
+    void generateDeathRegistrationGroupXmlGenerates25RecordsIfNumberOfRecordsNotSpecified() throws JsonProcessingException {
+        var event = new InsertDeathXmlRequest("Test", null);
+        when(xmlMapper.writeValueAsString(any())).thenReturn("XmlFile");
+
+        underTest.handleRequest(event, context);
+
+        verify(xmlMapper).writeValueAsString(argThat(a -> {
+            var group = (DeathRegistrationGroup) a;
+            return group.deathRegistrations().size() == 25;
+        }));
+        verify(awsService).putInBucket(eq(bucketName), any(), eq("XmlFile"));
+    }
+
+    @Test
+    void failingToWriteAsXmlThrowsError() throws JsonProcessingException {
+        var event = new InsertDeathXmlRequest("Test", 10);
+        when(xmlMapper.writeValueAsString(any())).thenThrow(mock(MappingException.class));
+
+        var exception = assertThrows(MappingException.class, () -> underTest.handleRequest(event, context));
+
+        assert (exception.toString().startsWith("Mock for MappingException"));
```

Comment:
could we change this to specify the mock in a var, and then assertEquals the exception.getCause()